File: gradingAssignment/grader.py
import sys
import re



class Grader:
    def __init__(self, numGrade) :
        self.g = numGrade
    def getGrade(self) :
        if self.g <= 100 and self.g >= 90 :
            return "A"
        elif self.g <= 89 and self.g >= 80 :
            return "B"
        elif self.g <= 79 and self.g >= 70 :
            return "C"
        elif self.g <= 69 and self.g >= 60 :
            return "D"
        else : return "F"

class Validate:
    def __init__(self, userInput) :
        self.g = userInput
    def validation(self) :
        if self.g > 100 or self.g < 0 :
            notAccurate = True
            print("Enter a number between 100 and 0.")
            return False
        else :
            return True

def output(sName, asgnmt, sGrade) :
    print("Student's Name: %s"%name + "\nAssignment: %s"%assignment + "\nLetter Grade: %s"%givenGrade )

name = raw_input("What is the student's name?: ")
assignment = raw_input("What is the name of the assignment?: ")
givenGrade = None

while givenGrade == None :
    numGrade = raw_input("What number grade did " + name + " receive on " + assignment+"?: ")
    try:
        int(numGrade)
        newValidation = Validate(int(numGrade))
        x = newValidation.validation()
        if x == False :
            givenGrade = None
        else:
            newGrader = Grader(int(numGrade))
            letterG = newGrader.getGrade()
            givenGrade = letterG
            output(name, assignment, givenGrade)
    except ValueError:
        print("Please enter a number between 100 and 0.")
        givenGrade = None

File: gradingAssignment/grader.rb
class Grader
    def initialize(numGrade)
        @num_grade = numGrade
    end
    def get_grades
        if @num_grade <= 100 and @num_grade >= 90
            return "A"
        elsif @num_grade <= 89 and @num_grade >= 80
            return "B"
        elsif @num_grade <= 79 and @num_grade >= 70
            return "C"
        elsif @num_grade <= 69 and @num_grade >= 60
            return "D"
        else
            return "F"
        end
    end
end

def output(sName, assignment, letterG)
    puts "Student's Name: #{sName}" + "Assignment: #{assignment}" + "Letter Grade: #{letterG}"
end

puts "Student's name: "
name = gets
puts "Name of the Assignment: "
asgnmt = gets
given_grade = "none"

while given_grade == "none"
    puts "What number grade did #{name} get on #{asgnmt}?: "
    numGrade = gets
    if Integer(numGrade, exception: false)
        given_grade = Integer(numGrade)
        grader = Grader.new(given_grade)
        x = grader.get_grades
        output(name, asgnmt, x)
    else
        given_grade = "none"
    end
end
